sql/src/main/codegen/includes/common.ftl

```diff
@@ -17,59 +17,53 @@
  * under the License.
  */
 
-// Using fully qualified name for Pair class, since Calcite also has a same class name being used in the Parser.jj
-org.apache.druid.java.util.common.Pair<Granularity, String> PartitionGranularity() :
+SqlNode PartitionGranularity() :
 {
   SqlNode e;
-  Granularity granularity;
-  String unparseString;
+  SqlNode result;
 }
 {
   (
     <HOUR>
     {
-      granularity = Granularities.HOUR;
-      unparseString = "HOUR";
+      result = SqlLiteral.createSymbol(GranularityType.HOUR, getPos());
     }
   |
     <DAY>
     {
-      granularity = Granularities.DAY;
-      unparseString = "DAY";
+      result = SqlLiteral.createSymbol(GranularityType.DAY, getPos());
     }
   |
     <MONTH>
     {
-      granularity = Granularities.MONTH;
-      unparseString = "MONTH";
+      result = SqlLiteral.createSymbol(GranularityType.MONTH, getPos());
     }
   |
     <YEAR>
     {
-      granularity = Granularities.YEAR;
-      unparseString = "YEAR";
+      result = SqlLiteral.createSymbol(GranularityType.YEAR, getPos());
     }
   |
     <ALL>
     {
-      granularity = Granularities.ALL;
-      unparseString = "ALL";
+      result = SqlLiteral.createSymbol(GranularityType.ALL, getPos());
     }
     [
       <TIME>
       {
-        unparseString += " TIME";
+        result = SqlLiteral.createSymbol(GranularityType.ALL, getPos());
```

Comment:
hmm, why is it `ALL` for `<TIME>`?

Reply:
ALL is for `ALL TIME`, this is the generated gode:
``` 
} else if (jj_2_65(2)) {
      jj_consume_token(ALL);
      result = SqlLiteral.createSymbol(GranularityType.ALL, getPos());
      if (jj_2_60(2)) {
        jj_consume_token(TIME);
        result = SqlLiteral.createSymbol(GranularityType.ALL, getPos());
      } else {
        ;
      }
    } 
```

Reply:
if they are the same - I think you could ignore the `TIME` keyword by:
```
<ALL> (<TIME>)?
```

Reply:
keeping the unparse string now, which is different in each case, so keeping them separate as before